### PR TITLE
Add a visualizer to see specific enemies depending on what filter is active

### PR DESCRIPTION
**Pull request by Corkhan**

### CODE REVIEW

File: Assets/FeatureVisualizeEnemies/VisualizeEnemies.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class VisualizeEnemies : MonoBehaviour {
+
+    private GameObject[] enemiesList;
+    public GameObject filtersSystem;
```
Comment (MrFouss) [inline Assets/FeatureVisualizeEnemies/VisualizeEnemies.cs]: Public properties should be in UpperCamelCase

File: Assets/FeatureVisualizeEnemies/VisualizeEnemies.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+                {
+                    enemy.GetComponent<Renderer>().enabled = false;
+                }
+            }
+        }
+		
+	}
+
+    // To be called whenever an enemy is created for it to be analyzed later on by the visualizer
+    public void updateEnemiesList()
```
Comment (MrFouss) [inline Assets/FeatureVisualizeEnemies/VisualizeEnemies.cs]: Methods should be in UpperCamelCase

File: Assets/FeatureVisualizeEnemies/VisualizeEnemies.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class VisualizeEnemies : MonoBehaviour {
+
+    private GameObject[] enemiesList;
+    public GameObject filtersSystem;
+    private FilterSelector filterSelector;
```
Comment (MrFouss) [inline Assets/FeatureVisualizeEnemies/VisualizeEnemies.cs]: The `FilterSelector` class doesn't exist, maybe do you mean `EnemyFilterSelector`?

File: Assets/FeatureVisualizeEnemies/EnemyFilterSelector.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+public class EnemyFilterSelector : MonoBehaviour {
+
+    // This script can be attached to enemies who require a special filter to be seen
+    // Attach it and a random filter will be attributed on Awake
+    // The EnemyVisualizer will take care of displaying the right ones
+    public int RequiredFilter;
+
+	void Awake() {
+        RequiredFilter = Random.Range(0, 3);
+        GameObject.Find("EnemyVisualizer").GetComponent<VisualizeEnemies>().updateEnemiesList();
```
Comment (MrFouss) [inline Assets/FeatureVisualizeEnemies/EnemyFilterSelector.cs]: If the EnemyVisualizer takes care of displaying the right enemies, why does your filter selector have to call explicitly a method from it? Maybe consider using events fired from here and catched by the visualizer (ask @Ezia for help, she'll help you with that).

**Review by MrFouss — CHANGES_REQUESTED** [review]

File: Assets/FeatureVisualizeEnemies/VisualizeEnemies.cs (hunk 3 above)
Comment (Corkhan) [reply Assets/FeatureVisualizeEnemies/VisualizeEnemies.cs]: I was referring to the FilterSelector in the FiltersSystem feature, which shows the correct filter according to the mouse wheel input.

File: Assets/FeatureVisualizeEnemies/EnemyFilterSelector.cs (hunk 4 above)
Comment (Corkhan) [reply Assets/FeatureVisualizeEnemies/EnemyFilterSelector.cs]: I will take care of that with her help tonight, thanks!

File: Assets/FeatureVisualizeEnemies/VisualizeEnemies.cs (hunk 2 above)
Comment (Corkhan) [reply Assets/FeatureVisualizeEnemies/VisualizeEnemies.cs]: This should be fixed.

File: Assets/FeatureVisualizeEnemies/VisualizeEnemies.cs (hunk 1 above)
Comment (Corkhan) [reply Assets/FeatureVisualizeEnemies/VisualizeEnemies.cs]: This should be fixed.

**Pull request merged**